resolver_test.go

```diff
@@ -2,14 +2,69 @@ package doh
 
 import (
 	"context"
+	"io/ioutil"
 	"net"
+	"net/http"
+	"net/http/httptest"
 	"testing"
+
+	"github.com/miekg/dns"
 )
 
+func mockDNSHeader(name string, rrType uint16) dns.RR_Header {
+	return dns.RR_Header{Name: name, Rrtype: rrType, Class: dns.ClassINET, Ttl: 300}
+}
+
+func mockDNSAnswerA(name string, ip net.IP) *dns.Msg {
+	m := new(dns.Msg)
+	m.Answer = make([]dns.RR, 1)
+	hdr := mockDNSHeader(name, dns.TypeA)
+	key := &dns.A{Hdr: hdr, A: ip}
+	m.Answer[0] = key
+	return m
+}
+
+func mockDNSAnswerAAAA(name string, ip net.IP) *dns.Msg {
+	m := new(dns.Msg)
+	m.Answer = make([]dns.RR, 1)
+	hdr := mockDNSHeader(name, dns.TypeAAAA)
+	key := &dns.AAAA{Hdr: hdr, AAAA: ip}
+	m.Answer[0] = key
+	return m
+}
+
+func mockDNSAnswerTXT(name string, records []string) *dns.Msg {
+	m := new(dns.Msg)
+	m.Answer = make([]dns.RR, 1)
+	hdr := mockDNSHeader(name, dns.TypeTXT)
+	key := &dns.TXT{Hdr: hdr, Txt: records}
+	m.Answer[0] = key
+	return m
+}
+
+func mockDoHResolver(msgs map[uint16]*dns.Msg) *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		body, _ := ioutil.ReadAll(req.Body)
```

Comment:
Don't ignore `error` values. If this fails at some point in the future, this would be quite hard to debug.
Better: pass `testing.T` to `mockDoHResolver`, and check the returned error.